src/init/starship.nu

```diff
@@ -2,37 +2,53 @@
 # - overlay which can be loaded with `overlay use starship.nu`
 # - module which can be used with `use starship.nu`
 # - script which can be used with `source starship.nu`
-export-env { $env.STARSHIP_SHELL = "nu"; load-env {
-    STARSHIP_SESSION_KEY: (random chars -l 16)
-    PROMPT_MULTILINE_INDICATOR: (
-        ^::STARSHIP:: prompt --continuation
-    )
+export-env {
+    $env.STARSHIP_SHELL = "nu"
 
-    # Does not play well with default character module.
-    # TODO: Also Use starship vi mode indicators?
-    PROMPT_INDICATOR: ""
+    $env.config.hooks.pre_execution ++= {||
+        let cmd = commandline | parse --regex '\s*(:?sudo\s+)?(?<cmd>\w+).*' | get cmd
```

Comment:
A basic heuristic to get the command - this works for `sudo cmd args` and `cmd args`.  For my case this is good enough, as it's mostly to avoid too many unneeded notifications.  It's fine if some commands are missed.